hopsworks-common/src/main/java/io/hops/hopsworks/common/serving/tf/LocalhostTfServingController.java

```diff
@@ -381,55 +224,78 @@ private void startTfServingInstance(Project project, Users user, TfServing tfSer
-        throw new TfServingException(RESTCodes.TfServingErrorCode.LIFECYCLEERRORINT, Level.INFO);
+        serving.setLocalPid(PID_STOPPED);
+        servingFacade.updateDbObject(serving, project);
+        throw new ServingException(RESTCodes.ServingErrorCode.LIFECYCLEERRORINT, Level.INFO);
       }
 
       // Read the pid for TensorFlow Serving server
       Path pidFilePath = Paths.get(secretDir.toString(), "tfserving.pid");
       String pidContents = Files.readFirstLine(pidFilePath.toFile(), Charset.defaultCharset());
 
       // Update the info in the db
-      tfServing.setLocalPid(Integer.valueOf(pidContents));
-      tfServing.setLocalPort(restPort);
-      tfServingFacade.updateDbObject(tfServing, project);
+      serving.setLocalPid(Integer.valueOf(pidContents));
+      serving.setLocalPort(restPort);
+      servingFacade.updateDbObject(serving, project);
     } catch (Exception ex) {
       // Startup process failed for some reason
-      tfServing.setLocalPid(PID_STOPPED);
-      tfServingFacade.updateDbObject(tfServing, project);
+      serving.setLocalPid(PID_STOPPED);
+      servingFacade.updateDbObject(serving, project);
 
-      throw new TfServingException(RESTCodes.TfServingErrorCode.LIFECYCLEERRORINT, Level.SEVERE, null,
+      throw new ServingException(RESTCodes.ServingErrorCode.LIFECYCLEERRORINT, Level.SEVERE, null,
           ex.getMessage(), ex);
 
     } finally {
       if (settings.getHopsRpcTls()) {
         certificateMaterializer.removeCertificatesLocal(user.getUsername(), project.getName());
       }
-      // release lock on the tfServing entry
-      tfServingFacade.releaseLock(project, tfServing.getId());
+      // release lock on the serving entry
+      servingFacade.releaseLock(project, serving.getId());
+    }
+  }
+  
+  /**
+   * Validates user data for creating or updating a Tensorflow Serving Instance
+   *
+   * @param servingWrapper the user data
+   * @param project the project to create the serving for
+   * @throws ServingException if the python environment is not activated for the project
+   */
+  public void validateUserInput(ServingWrapper servingWrapper, Project project) {
+    // Check that the modelPath respects the TensorFlow standard
+    validateModelPath(servingWrapper.getServing().getArtifactPath(),
+      servingWrapper.getServing().getVersion());
+    
+    // Check that the batching option has been specified
+    if (servingWrapper.getServing().isBatchingEnabled() == null) {
+      throw new IllegalArgumentException("Batching is null");
     }
   }
 
-  private void restartTfServingInstance(Project project, Users user, TfServing currentInstance,
-                                        TfServing newInstance) throws TfServingException {
-    // Kill current TfServing instance
-    killTfServingInstance(project, currentInstance, false);
+  private void validateModelPath(String path, Integer version) throws IllegalArgumentException {
+    try {
```

Comment:
Same thing here. Any way we can avoid code duplication between the Localhost and K8s version?

Reply:
Done, moved common stuff to a Util Bean